### PR TITLE
✨ [RUMF-1084] ignore init if a RUM instance is or will be injected by synthetics

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Allows the Synthetics worker to prevent any RUM SDK initialisation if it plans to inject a RUM instance itself.

## Changes

* Move synthetics context in its own module
* Ignore `init()` call based on values (global variables/cookies) provided by the Synthetics worker

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 1]
```diff
@@ -103,7 +104,7 @@ export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: S
       return
     }
 
-    if (!isValidInitConfiguration(initConfiguration)) {
+    if (willSyntheticsInjectRum() || !isValidInitConfiguration(initConfiguration)) {
```
Comment (amortemousque) [inline packages/rum-core/src/boot/rumPublicApi.ts:107]: By adding the check here, we still check `canHandleSession`. So we might have init cookie error message even if the SDK won't be started.

File: packages/rum-core/src/tools/syntheticsContext.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { getCookie } from '@datadog/browser-core'
```
Comment (amortemousque) [inline packages/rum-core/src/tools/syntheticsContext.ts:1]: `tools/syntheticsContext.ts`, should it be consider as a tool?

Comment (BenoitZugmeyer) [reply packages/rum-core/src/tools/syntheticsContext.ts:1]: Would you rather put it in `domain`? I thought it wasn't directly related to how RUM works, but happy to move it elsewhere.

File: packages/rum-core/src/boot/rumPublicApi.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply packages/rum-core/src/boot/rumPublicApi.ts:107]: Good point, I'll move it

File: packages/rum-core/src/tools/syntheticsContext.ts (hunk 2 above)
Comment (amortemousque) [reply packages/rum-core/src/tools/syntheticsContext.ts:1]: Why not to have tools, but I would also think of moving other components like `ContextHistory` which also looks like tools.

Comment (bcaudan) [reply packages/rum-core/src/tools/syntheticsContext.ts:1]: IMO
- syntheticsContext -> domain
- contextHistory -> tools

File: packages/rum-core/src/tools/syntheticsContext.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    expect(getSyntheticsContext()).toBeUndefined()
+  })
+})
+
+describe('willSyntheticsInjectRum', () => {
+  afterEach(() => {
+    cleanupSyntheticsWorkerValues()
+  })
+
+  it('returns false if nothing is defined"', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/tools/syntheticsContext.spec.ts:51]:
```suggestion
  it('returns false if nothing is defined', () => {
```

File: packages/rum-core/src/tools/syntheticsContext.spec.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    mockSyntheticsWorkerValues({}, 'globals')
+
+    expect(willSyntheticsInjectRum()).toBeFalse()
+  })
+
+  it('returns false if the INJECTS_RUM global variable is false"', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/tools/syntheticsContext.spec.ts:57]:
```suggestion
  it('returns false if the INJECTS_RUM global variable is false', () => {
```

File: packages/rum-core/src/tools/syntheticsContext.spec.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    mockSyntheticsWorkerValues({ injectsRum: false }, 'globals')
+
+    expect(willSyntheticsInjectRum()).toBeFalse()
+  })
+
+  it('returns true if the INJECTS_RUM global variable is truthy"', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/tools/syntheticsContext.spec.ts:63]:
```suggestion
  it('returns true if the INJECTS_RUM global variable is truthy', () => {
```

File: packages/rum-core/src/tools/syntheticsContext.spec.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    mockSyntheticsWorkerValues({ injectsRum: true }, 'globals')
+
+    expect(willSyntheticsInjectRum()).toBeTrue()
+  })
+
+  it('returns true if the INJECTS_RUM cookie is truthy"', () => {
```
Comment (bcaudan) [inline packages/rum-core/src/tools/syntheticsContext.spec.ts:69]:
```suggestion
  it('returns true if the INJECTS_RUM cookie is truthy', () => {
```

**Review by bcaudan — APPROVED** [review]

**Review by amortemousque — APPROVED** [review]

**Pull request merged**